go/util/datetime/date_time.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Package datetime implements marshalling of Go DateTime values into Noms structs
+// with type DateTimeType.
+package datetime
+
+import (
+	"math"
+	"time"
+
+	"github.com/attic-labs/noms/go/marshal"
+	"github.com/attic-labs/noms/go/types"
+)
+
+// DateTime is an alias for time.Time that allows us to marshal date time to
+// Noms.
+type DateTime time.Time
+
+// DateTimeType is the Noms type used to represent date time objects in Noms.
```

Comment:
Might want to indicate somewhere that it is valid for secSinceEpoch to be fractional.